pallets/creditcoin/src/lib.rs

```diff
@@ -806,6 +823,13 @@ pub mod pallet {
 			let address = Self::get_address(&address_id)?;
 			ensure!(address.owner == who, Error::<T>::NotAddressOwner);
 
+			let currency =
+				Currencies::<T>::get(&terms.currency).ok_or(Error::<T>::CurrencyNotRegistered)?;
+			ensure!(
+				address.blockchain == currency.blockchain(),
+				Error::<T>::AddressPlatformMismatch
+			);
```

Comment:
Similar comments as for `add_ask_order`. Looks like this new snippet is missing test coverage.